fate/src/main/java/org/apache/accumulo/fate/zookeeper/Retry.java

```diff
@@ -166,18 +166,26 @@ protected void sleep(long wait) throws InterruptedException {
   }
 
   public void logRetry(Logger log, String message, Throwable t) {
-    // log the first time, and then after every logInterval
-    if (lastRetryLog < 0 || (System.currentTimeMillis() - lastRetryLog) > logInterval) {
+    // log the first time as debug, and then after every logInterval as a warning
+    long now = System.currentTimeMillis();
+    if (lastRetryLog < 0) {
+      log.debug(getMessage(message));
+      lastRetryLog = now;
+    } else if ((now - lastRetryLog) > logInterval) {
       log.warn(getMessage(message), t);
-      lastRetryLog = System.currentTimeMillis();
+      lastRetryLog = now;
     }
   }
 
   public void logRetry(Logger log, String message) {
-    // log the first time, and then after every logInterval
-    if (lastRetryLog < 0 || (System.currentTimeMillis() - lastRetryLog) > logInterval) {
+    // log the first time as debug, and then after every logInterval as a warning
+    long now = System.currentTimeMillis();
```

Comment:
I strongly recommend using `System.nanoTime()` for elapsed time. `System.currentTimeMillis()` is for clock time, which can go backwards and result in a surprising negative difference between the two values.

Reply:
ok, will do